app/shared/app-data/src/androidMain/kotlin/domain/torrent/client/RemoteCall.kt

```diff
@@ -72,4 +96,40 @@ class RetryRemoteCall<I : IInterface>(
             null
         }
     }
+}
+
+/**
+ * Wrapper for call which takes a continuation-like argument and returns [IDisposableHandle],
+ * which means this is a asynchronous RPC call.
+ *
+ * [IDisposableHandle] takes responsibility to pass cancellation to server.
+ */
+suspend inline fun <I : IInterface, T> RemoteCall<I>.callSuspendCancellable(
+    crossinline transact: I.(
+        resolve: (T?) -> Unit,
+        reject: (RemoteContinuationException?) -> Unit
+    ) -> IDisposableHandle?,
+): T = suspendCancellableCoroutine { cont ->
+    val disposable = call {
+        transact(
+            { value ->
+                if (value == null) {
+                    cont.resumeWithException(CancellationException("Remote resume a null value."))
```

Comment:
```suggestion
                    cont.resumeWithException(CancellationException("Remote resume a null value."))
```

CancellationException 语义正确吗?

自己构造 CancellationException 通常来说是错误的

Reply:
这里可能要用 IllegalStateException?

Reply:
好像确实是